src/app/components/Ionic/IonicProvider.tsx

```diff
@@ -0,0 +1,38 @@
+import { createContext, FC, PropsWithChildren, useEffect } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { Capacitor } from '@capacitor/core'
+import { App } from '@capacitor/app'
+
+const IonicContext = createContext<undefined>(undefined)
+
+const IonicContextProvider: FC<PropsWithChildren> = ({ children }) => {
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    /**
+     * The back button refers to the physical back button on an Android device and should not be confused
+     * with either the browser back button or ion-back-button.
+     */
+    App.addListener('backButton', ({ canGoBack }) => {
+      if (!canGoBack) {
+        App.exitApp()
```

Comment:
shoudn't we call `minimizeApp` here?

Reply:
Not sure exactly what minimizeApp does. But exitApp actually minimizes the app, and it behaves like any other app on Android at least. This was taken from example anyhow.